### PR TITLE
Update cross_staff_offset.lua

### PR DESCRIPTION
**Pull request by cv-on-hub**
Re-jigged for modal position-retained window, implemented shift-alt option to suppress dialog. Sorry about all the merge branch commits. GitHub Desktop made me do it.

### CODE REVIEW

**Review by rpatters1 — APPROVED** [review]
The file looks fine to me. I'll let @Nick-Mazuk merge this one to confirm that the merged branch history is not a problem.

File: src/cross_staff_offset.lua [hunk 1]
```diff
@@ -4,154 +4,132 @@ function plugindef()
...
     finaleplugin.AuthorURL = "http://carlvine.com/lua/"
-    finaleplugin.Version = "v1.29"
-    finaleplugin.Date = "2022/07/11"
+    finaleplugin.Version = "v1.33"
+    finaleplugin.Date = "2022/08/05"
 	finaleplugin.MinJWLuaVersion = 0.62
-    finaleplugin.AdditionalMenuOptions = [[  CrossStaff Offset No Dialog  ]]
-    finaleplugin.AdditionalUndoText = [[     CrossStaff Offset No Dialog  ]]
-    finaleplugin.AdditionalPrefixes = [[     no_user_dialog = true  ]]
-    finaleplugin.AdditionalDescriptions = [[ Offset horizontal position of cross-staff note entries - NO DIALOG ]]
```
Comment (Nick-Mazuk) [inline src/cross_staff_offset.lua:13]: Looks like as is this PR is removing the additional menu options?

**Comment by Nick-Mazuk** [comment]
In general, rebasing is preferred over merges. Merge commits aren't inherently bad or problematic, but they have the tendency to cause problems. This looks like a place where the merge commits may be causing problems (e.g., then menu options look deleted when I'm guessing you didn't delete them).

Here's a fairly good guide about the two: https://www.atlassian.com/git/tutorials/merging-vs-rebasing

I honestly don't know why GitHub pushes merge commits so much (no pun intended).

In short, I don't have a problem with merge commits entering into this repo as long as they aren't causing issues (we've done this many times in the past). But if they cause problems, I'd rather them not enter the repo.

**Comment by rpatters1** [comment]
But does it matter if it's coming in from a forked repo? Basically the only merge history the source repo should have would be in the comments. Is that not correct?

File: src/cross_staff_offset.lua (hunk 1 above)
Comment (cv-on-hub) [reply src/cross_staff_offset.lua:13]: > Looks like as is this PR is removing the additional menu options?

Yes it is - following on from various discussions that it was more consistent to offer **No Dialog** behaviour via modifier key action that's exactly what this script introduces. (As well as retaining dialog position with RGP 0.64+)

**Comment by cv-on-hub** [comment]
> Here's a fairly good guide about the two:

Thanks for the info @Nick-Mazuk. I've tried to read through most of the introductory stuff in that guide, and it seems that what I should do is: 
1. work on new features in a new branch ("feature")
2. create a PR
3. after the PR is approved I **rebase** `main` before merging "feature" back into `main`

Is that right?

**Comment by rpatters1** [comment]
Another thought along these lines is that Github offers a "squash and merge" option to merge all the commits from the fork in a single commit in the master branch. (Click the down arrow next to the `Merge pull request` button.) I don't personally like this option because it cuts off the connection to the main branch from my fork branch in the visualization in Sourcetree. I like seeing my fork branch flow back into the main branch. But ymmv. I'm open to embracing it.

**Comment by cv-on-hub** [comment]
I await your conclusive collective wisdom.
